modules/tinymce/src/plugins/table/main/ts/ui/CellDialog.ts

```diff
@@ -49,12 +56,26 @@ const applyCellData = (editor: Editor, cells: HTMLTableCellElement[], data: Cell
   const dom = editor.dom;
   const isSingleCell = cells.length === 1;
 
-  Arr.each(cells, (cell) => {
+  const table = TableLookup.table(SugarElement.fromDom(cells[0])).getOrNull();
```

Comment:
This is not safe, as this will throw an exception as `fromTable` doesn't allow a null value. A the very least, this should be `getOrDie`, however can this be refactored to avoid that case?

Reply:
This hasn't been resolved or even changed at all... so unresolving.

Reply:
If you look a line up, we now check that the cells has at least one entry in it, so it will not be null. We also check that the table is not null, which works on the apply.

Reply:
That's not the functional way of doing things... generally if you're having to use `getOrNull` or `getOrDie` than something is wrong. You also are only doing that in 1 place, not both places (which again goes back to my comment above removing the duplication).

Reply:
As discussed today, this should use something like this instead (or something similar):
```
TableLookup.table(SugarElement.fromDom(cells[0])).each(...)
```